doc/api/repl.md

```diff
@@ -2,80 +2,129 @@
-dropped into the REPL. It has simplistic emacs line-editing.
+```js
+const repl = require('repl');
+```
+
+## Design and Features
+
+The `repl` module exports the `repl.REPLServer` class. While running, instances
+of `repl.REPLServer` will accept individual lines of user input, evaluate those
+according to a user-defined evaluation function, then output the result. Input
+and output may be from `stdin` and `stdout`, respectively, or may be connected
+to any Node.js [stream][].
+
+Instances of `repl.REPLServer` support persistent history of prior inputs,
+automatic completion of inputs, simplistic Emacs-style line editing, multi-line
+inputs, ANSI-styled output, saving and restoring current REPL session state,
+error recovery, and customizable evaluation functions.
+
+### Commands and Special Keys
+
+The following special commands are supported by all REPL instances:
+
+* `.break` - When in the process of inputting a multi-line expression, entering
+  the `.break` command (or pressing the `<Ctrl>-C` key combination) will abort
+  further input or processing of that expression.
+* `.clear` - Resets the REPL `context` to an empty object and clears any
+  multi-line expression currently being input.
+* `.exit` - Close the I/O stream, causing the REPL to exit.
+* `.help` - Show this list of special commands.
+* `.save` - Save the current REPL session to a file:
+  `>.save ./file/to/save.js`
+* `.load` - Load a file into the current REPL session.
+  `>.load ./file/to/load.js`
+
+The following key combinations in the REPL have these special effects:
+
+* `<ctrl>-C` - When pressed once, has the same effect as the `.break` command.
+  When pressed twice on a blank line, has the same effect as the `.exit`
+  command.
+* `<ctrl>-D` - Has the same effect as the `.exit` command.
+* `<tab>` - When pressed on a blank line, displays global and local(scope)
+  variables. When pressed while entering other input, displays relevant
+  autocompletion options.
+
+### Default Evaluation
+
+By default, all instances of `repl.REPLServer` use an evaluation function that
+evaluates JavaScript expressions and provides access to Node.js' built-in
+modules. This default behavior can be overridden by passing in an alternative
+evaluation function when the `repl.REPLServer` instance is created.
+
+#### JavaScript Expressions
+
+The default evaluator supports direct evaluation of JavaScript expressions:
 
 ```
-$ node
-Type '.help' for options.
-> a = [1, 2, 3];
-[ 1, 2, 3 ]
-> a.forEach((v) => {
-...   console.log(v);
-...   });
-1
+> 1 + 1
 2
+> var m = 2
+1
+> m + 1
 3
 ```
 
-For advanced line-editors, start Node.js with the environmental variable
-`NODE_NO_READLINE=1`. This will start the main and debugger REPL in canonical
-terminal settings which will allow you to use with `rlwrap`.
+Unless otherwise scoped within blocks (e.g. `{ ... }`) or functions, variables
+declared either implicitly or using the `var` keyword are declared at the
+`global` scope.
```

Comment:
@lance just fyi, #5703 would be changing this… so if this PR lands first (which I guess it will?), you may want to add a change to the docs to your PR